packages/react-input/src/components/Input/Input.types.ts

```diff
@@ -21,7 +27,14 @@ export type InputSlots = {
   contentAfter?: IntrinsicShorthandProps<'span'>;
 };
 
-export type InputCommons = {
+/**
+ * Input Props
+ */
+export type InputProps = Omit<
+  ComponentProps<InputSlots, 'input'>,
+  // `children` is unsupported. The rest of these native props have customized definitions.
```

Comment:
Does omitting `children` actually work? I think React.forwardRef might add back in `children` to the final props type?

Instead of omitting it, you might declare it as e.g.:
```ts
/**
 * Input can't have children
 */
children?: undefined;
```